test/libsolidity/util/BytesUtils.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/util/BytesUtils.h>
+
+#include <liblangutil/Common.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <fstream>
+#include <memory>
+#include <regex>
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+using namespace soltest;
+
+bytes BytesUtils::convertBoolean(string const& _literal)
+{
+	if (_literal == "true")
+		return bytes{true};
```

Comment:
Why is it just a single byte?

Reply:
That's because of the alignment that will be applied later on. You could write `f() -> left(bool)` otherwise the default right-alignment will be used.

Reply:
OK, but I would say it may be better to force the user to use `left(0x10)` in that case, because what is a left-aligned or right-aligned bool - but we can also keep it for now.